lib/reverse_words.rb

```diff
@@ -1,6 +1,26 @@
-# A method to reverse each word in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time Complexity: O(n) because there is a constant ratio (3n/2) to the size of the input array.
+# Space Complexity: O(1) because no additional memory is needed regardless of input array.
+
 def reverse_words(my_words)
-  raise NotImplementedError
+  i = 0
+  j = 0
+  while my_words && j <= my_words.length
+    if my_words[j] == ' ' || j == my_words.length
+      little_end = i
+      big_end = j - 1
+      reverse_substring(my_words, little_end, big_end)
+      i = j + 1
+    end
+    j += 1
+  end
+end
+
+def reverse_substring(my_words, little_end, big_end)
+  while little_end < big_end
+    pushpin = my_words[little_end]
```

Comment:
_Super_-tiny nit: conventionally, these `pushpin`-type variables are either called or prefixed with `temp`.

Conventions like these can make code easier to for others to read at a glance. 🙂

Reply:
Brilliant! Thank you. I never know what to call these things. "temp" makes a heck of a lot more sense. Learning conventions is a relief -- less time spent trying to come up with readable variable names and more time to actually *do* something with them. :)